File: 01-arrays_&_hashing/00242-Valid_Anagram/00242-valid_anagram.go
// 242: Valid Anagram
// https://leetcode.com/problems/valid-anagram/

package main

import "fmt"

func isAnagram(s, t string) bool {
    if len(s) != len(t) { return false }

    n := len(s)
    counts := [26]int{}

    for i:=0; i<n; i++ {
        counts[s[i] - 'a']++
        counts[t[i] - 'a']--
    }

    for i:=0; i<26; i++ {
        if counts[i] != 0 {return false}
    }

    return true
}

func main() {
	// INPUT
	s := "rat"
    t := "cat"

	// OUTPUT
    result := isAnagram(s, t)
    fmt.Println(result)
}

File: 01-arrays_&_hashing/00242-Valid_Anagram/00242-valid_anagram.py
# 242: Valid Anagram
# https://leetcode.com/problems/valid-anagram/

from itertools import count


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t) : return False

        n = len(s)
        counts = [0]*26

        for i in range(n):
            counts[ord(s[i]) - ord('a')]+=1
            counts[ord(t[i]) - ord('a')]-=1

        for i in range(26):
            if counts[i] != 0 : return False
        
        return True


if __name__ == "__main__":
    o = Solution()

    # INPUT
    s: str = "rat"
    t: str = "cat"

    # OUTPUT
    result = o.isAnagram(s, t)
    print(result)
